base_layer/core/src/core_key_manager/service.rs

```diff
@@ -626,11 +660,22 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
         total_excess: &PublicKey,
         kernel_version: &TransactionKernelVersion,
         kernel_message: &[u8; 32],
+        kernel_features: &KernelFeatures,
+        txo_type: TxoType,
     ) -> Result<Signature, TransactionError> {
-        let spending_private_key = self.get_private_key(spending_key).await?;
+        let spending_private_key = if txo_type == TxoType::Output {
+            self.get_private_key(spending_key).await?
+        } else {
+            PrivateKey::default() - self.get_private_key(spending_key).await?
+        };
         let private_nonce = self.get_private_key(nonce_id).await?;
-        let signing_key =
-            spending_private_key - &self.get_partial_private_kernel_offset(spending_key, nonce_id).await?;
+        // we cannot use an offset with a coinbase tx as this will not allow us to check the coinbase commitment
+        let signing_key = if kernel_features.is_coinbase() {
+            spending_private_key
+        } else {
+            spending_private_key - &self.get_partial_private_kernel_offset(spending_key, nonce_id).await?
```

Comment:
would this not be 0 anyway?

Reply:
For kernels it needs to be 0. 
But the function to calculate the offset is oblivious to the context that is used to calculate the offset. 
It just knows that given two id's, it will calculate an offset. 

So we need to not call it if we sign a coinbase kernel.

Reply:
alright, not 100% convinced, but ok